### PR TITLE
GUI: tabs with scrollbars and move game settings out of Engine tab

### PR DESCRIPTION
**Pull request by Die4Ever**
image

### CODE REVIEW

File: gui/widgets/tab.cpp [hunk 1]
```diff
@@ -122,6 +121,7 @@ int TabWidget::addTab(const Common::U32String &title, const Common::String &dial
 	newTab.title = title;
```
Comment (Die4Ever) [inline gui/widgets/tab.cpp:121]: we might want to add a bool withScroll argument to addTab, so you can create tabs without the automatic ScrollContainerWidget, or else we will have to fix all the old tabs that used scrollbars like GameOptions_KeyMapper

I do want to figure out why the nested scrollbars aren't working properly before I do either of those above things

File: gui/object.cpp [hunk 2]
```diff
@@ -59,6 +59,12 @@ void GuiObject::resize(int x, int y, int w, int h, bool scale) {
 	}
 }
 
+Widget *GuiObject::addChild(Widget *newChild) {
```
Comment (Die4Ever) [inline gui/object.cpp:62]: maybe we should call setNext here instead of trusting the caller (Widget::init) to handle that?

File: gui/widget.cpp [hunk 3]
```diff
@@ -51,9 +51,7 @@ Widget::Widget(GuiObject *boss, const Common::String &name, const Common::U32Str
 }
 
 void Widget::init() {
-	// Insert into the widget list of the boss
-	_next = _boss->_firstWidget;
-	_boss->_firstWidget = this;
+	_next = _boss->addChild(this);
```
Comment (Die4Ever) [inline gui/widget.cpp:54]: maybe the addChild function should call setNext on us, instead of doing it here?

File: gui/widget.cpp [hunk 4]
```diff
@@ -119,10 +118,8 @@ void Widget::draw() {
 				_x = _x + g_gui.getOverlayOffset();
 			}
 
-			Common::Rect r = _boss->getClipRect();
-			r.moveTo(_x, r.top);
-
-			g_gui.theme()->swapClipRect(r);
+			clip.moveTo(_x, clip.top);
+			g_gui.theme()->swapClipRect(clip);
```
Comment (Die4Ever) [inline gui/widget.cpp:122]: I'm not entirely sure about this change, but it appears to work with Hebrew and English, and both the old and new style scrollcontainers in tabs

File: gui/widget.cpp (hunk 3 above)
Comment (bluegr) [reply gui/widget.cpp:54]: Could be done as a future enhancement

File: gui/widget.cpp (hunk 4 above)
Comment (bluegr) [reply gui/widget.cpp:122]: This code is used in RTL languages. Since it works with Hebrew, it should be fine

**Comment by bluegr** [comment]
Nice work overall! Moving the game options to the game tab does make more sense indeed. I've initially added them in a separate tab, because back then we didn't have the functionality we got now in the GUI.

Overall, the code is quite extensible and reusable, and the functionality will hopefully help us for future additions as well.

Thanks again! Merging

**Pull request merged**